src/Microsoft.Data.SqlClient/netcore/src/Microsoft/Data/SqlClient/TdsParser.cs

```diff
@@ -3073,51 +3073,63 @@ private TdsOperationStatus TryProcessEnvChange(int tokenLength, TdsParserStateOb
                         break;
 
                     case TdsEnums.ENV_ROUTING:
-                        ushort newLength;
-                        result = stateObj.TryReadUInt16(out newLength);
-                        if (result != TdsOperationStatus.Done)
-                        {
-                            return result;
-                        }
-                        env._newLength = newLength;
-                        byte protocol;
-                        result = stateObj.TryReadByte(out protocol);
-                        if (result != TdsOperationStatus.Done)
-                        {
-                            return result;
-                        }
-                        ushort port;
-                        result = stateObj.TryReadUInt16(out port);
-                        if (result != TdsOperationStatus.Done)
-                        {
-                            return result;
-                        }
-                        ushort serverLen;
-                        result = stateObj.TryReadUInt16(out serverLen);
-                        if (result != TdsOperationStatus.Done)
-                        {
-                            return result;
-                        }
-                        string serverName;
-                        result = stateObj.TryReadString(serverLen, out serverName);
-                        if (result != TdsOperationStatus.Done)
                         {
-                            return result;
-                        }
-                        env._newRoutingInfo = new RoutingInfo(protocol, port, serverName);
-                        ushort oldLength;
-                        result = stateObj.TryReadUInt16(out oldLength);
-                        if (result != TdsOperationStatus.Done)
-                        {
-                            return result;
+                            ushort newLength;
```

Comment:
So the whole deal with this big block o changes was you added scope `{}` to this case? ... If we can avoid doing that that'd be better. But, if you reallly reallllly want to use a new variable in two scopes, I'll let it slide